# ✅ Solved Questions — Class 3
## User Input, Type Conversion, and Operators

This notebook contains **all 30 practice questions from Class 3**, solved in order, section by section.

> **Note on `input()` questions:** Several questions ask you to get input from the user. The solutions below use `input()` exactly as a real program would. When you run these cells yourself in Jupyter, they will pause and ask you to type something — that is expected behavior, not an error.

# Section 1 — Practice: Strings

### Question 1 — Create a variable named `name` and store your name in it.

In [1]:
name = "Saad"
print(name)

Saad


**Line-by-line explanation**
1. `name = "Saad"` creates a String variable holding a name.
2. `print(name)` displays it.

### Question 2 — Create two string variables and print both of them.

In [2]:
city = "Karachi"
country = "Pakistan"

print(city)
print(country)

Karachi
Pakistan


**Line-by-line explanation**
1. `city` and `country` are created as two separate String variables.
2. The two `print()` calls display each value on its own line.

### Question 3 — Create a string containing your city and print it.

In [3]:
city = "Karachi"
print(city)

Karachi


**Line-by-line explanation**
1. `city = "Karachi"` stores a city name as a String.
2. `print(city)` displays `Karachi`.

### Question 4 — Create a string containing a sentence with spaces and print it.

In [4]:
sentence = "I am learning Python step by step."
print(sentence)

I am learning Python step by step.


**Line-by-line explanation**
1. A String can contain spaces just like any other character — Python does not treat spaces specially inside quotes.
2. `print(sentence)` displays the full sentence exactly as written.

### Question 5 — Create three string variables and print them as separate lines.

In [5]:
line1 = "This is the first line."
line2 = "This is the second line."
line3 = "This is the third line."

print(line1)
print(line2)
print(line3)

This is the first line.
This is the second line.
This is the third line.


**Line-by-line explanation**
1. Three separate String variables are created, one per sentence.
2. Each `print()` call automatically starts a new line afterward, so the three sentences appear as three separate lines of output.

# Section 2 — Practice: User Input

### Question 1 — Ask the user for their name and print it.

In [6]:
name = input("What is your name? ")
print(name)

Saad


**Line-by-line explanation**
1. `input("What is your name? ")` displays the prompt text and pauses the program, waiting for the user to type something and press Enter.
2. Whatever the user types is returned as a **String** and stored in `name`.
3. `print(name)` displays whatever the user entered.

### Question 2 — Ask the user for their favorite color and print a sentence containing it.

In [9]:
color = input("What is your favorite color? ")
print("Your favorite color is " + color + ".")

Your favorite color is Teal.


**Line-by-line explanation**
1. `input(...)` collects the user's favorite color as text.
2. `"Your favorite color is " + color + "."` joins three String pieces with `+` — the label, the user's answer, and a closing period — into one full sentence.
3. `print(...)` displays that sentence.

### Question 3 — Ask the user for their city and print the city.

In [10]:
city = input("Which city are you from? ")
print(city)

Karachi


**Line-by-line explanation**
1. `input(...)` collects the city name as text.
2. `print(city)` displays exactly what the user typed.

### Question 4 — Ask the user for two pieces of text and print both.

In [12]:
favorite_food = input("What is your favorite food? ")
favorite_movie = input("What is your favorite movie? ")

print(favorite_food)
print(favorite_movie)

Beef steak
Pacific Rim


**Line-by-line explanation**
1. Two separate `input()` calls collect two separate pieces of text, one after the other.
2. Each `print()` call displays one of the collected values, on its own line.

### Question 5 — Ask the user for their name and city, then print a sentence containing both.

In [13]:
name = input("What is your name? ")
city = input("Which city are you from? ")

print(name + " is from " + city + ".")

Saad is from Karachi.


**Line-by-line explanation**
1. Two `input()` calls collect the name and city as text.
2. `+` joins the pieces (name, the fixed text `" is from "`, city, and a closing period) into a single sentence.
3. `print(...)` displays the combined sentence, e.g. `Saad is from Karachi.`

# Section 3 — Practice: Type Conversion

### Question 1 — Ask the user for a number, convert it to an integer, and print it.

In [15]:
age_text = input("Enter your age: ")
age = int(age_text)

print(age)

18


**Line-by-line explanation**
1. `input(...)` always returns text (a String), even if the user types digits — so `age_text` holds something like `"25"`, not the number `25`.
2. `int(age_text)` converts that text into a real Integer.
3. `print(age)` displays the number, now usable in math (e.g. `age + 1` would work; `age_text + 1` would not).

### Question 2 — Ask for two integers and print their sum.

In [16]:
num1 = int(input("Enter the first number: "))
num2 = int(input("Enter the second number: "))

print(num1 + num2)

1332


**Line-by-line explanation**
1. `int(input(...))` converts the user's text input into an Integer in one step, combining Question 1's two lines into one.
2. This is done twice, once for each number.
3. `num1 + num2` adds the two Integers together, and `print(...)` displays the sum.

### Question 3 — Ask for a decimal number, convert it to `float`, and print it.

In [17]:
weight_text = input("Enter your weight: ")
weight = float(weight_text)

print(weight)

64.0


**Line-by-line explanation**
1. `input(...)` returns text, e.g. `"60.5"`.
2. `float(weight_text)` converts that text into a real Float, `60.5`.
3. `print(weight)` displays the number.

### Question 4 — Ask for a number and print its type after converting it to `int`.

In [18]:
number = int(input("Enter a whole number: "))
print(type(number))

<class 'int'>


**Line-by-line explanation**
1. `int(input(...))` reads the input and immediately converts it to an Integer.
2. `type(number)` confirms the conversion worked by displaying `<class 'int'>`.

### Question 5 — Ask for a number as input, convert it to an integer, and calculate its double.

In [19]:
number = int(input("Enter a number: "))
double = number * 2

print(double)

18


**Line-by-line explanation**
1. `int(input(...))` gets the number and converts it to an Integer, stored in `number`.
2. `number * 2` calculates double the value.
3. `print(double)` displays the doubled result.

# Section 4 — Practice: Operators

### Question 1 — Create two numbers and calculate their sum.

In [20]:
a = 15
b = 8

total = a + b
print(total)

23


**Line-by-line explanation**
1. `a = 15` and `b = 8` create two numbers.
2. `a + b` adds them together, giving `23`.
3. `print(total)` displays `23`.

### Question 2 — Create two numbers and calculate their difference and product.

In [21]:
a = 15
b = 8

difference = a - b
product = a * b

print(difference)
print(product)

7
120


**Line-by-line explanation**
1. `a - b` calculates the difference: `15 - 8 = 7`.
2. `a * b` calculates the product: `15 * 8 = 120`.
3. Both results are printed on separate lines.

### Question 3 — Create two numbers and calculate their normal division.

In [22]:
a = 15
b = 8

division = a / b
print(division)

1.875


**Line-by-line explanation**
1. `a / b` performs normal (true) division: `15 / 8 = 1.875`.
2. `print(division)` displays `1.875` — note the result is always a Float, even though both inputs were Integers.

### Question 4 — Create two numbers and find the remainder using `%`.

In [23]:
a = 15
b = 8

remainder = a % b
print(remainder)

7


**Line-by-line explanation**
1. `a % b` finds the remainder of `15 ÷ 8`. Since `8 × 1 = 8` and `15 - 8 = 7`, the remainder is `7`.
2. `print(remainder)` displays `7`.

### Question 5 — Create a number and calculate its square using `**`.

In [24]:
number = 9
square = number ** 2

print(square)

81


**Line-by-line explanation**
1. `number ** 2` raises `number` to the power `2`, which is the mathematical definition of "squaring" a number: `9 * 9 = 81`.
2. `print(square)` displays `81`.

# Section 5 — Practice: Comparison Operators

### Question 1 — Create an `age` variable and check whether it is equal to 18.

In [25]:
age = 18
print(age == 18)

True


**Line-by-line explanation**
1. `age = 18` sets the value.
2. `age == 18` checks equality; since `age` is exactly `18`, this is `True`.
3. `print(...)` displays `True`.

### Question 2 — Create two numbers and check whether the first is greater than the second.

In [26]:
a = 20
b = 12

print(a > b)

True


**Line-by-line explanation**
1. `a > b` checks if `20` is strictly greater than `12`, which is `True`.
2. `print(...)` displays `True`.

### Question 3 — Create two numbers and check whether they are not equal.

In [27]:
a = 20
b = 12

print(a != b)

True


**Line-by-line explanation**
1. `a != b` checks "not equal". Since `20` and `12` are different, this is `True`.
2. `print(...)` displays `True`.

### Question 4 — Create a number and test whether it is greater than or equal to 50.

In [28]:
number = 50
print(number >= 50)

True


**Line-by-line explanation**
1. `number >= 50` checks greater-than-or-equal. Since `number` is exactly `50`, this is `True`.
2. `print(...)` displays `True`.

### Question 5 — Create a number and test whether it is less than or equal to 100.

In [29]:
number = 75
print(number <= 100)

True


**Line-by-line explanation**
1. `number <= 100` checks less-than-or-equal. Since `75` is indeed less than `100`, this is `True`.
2. `print(...)` displays `True`.

# Section 6 — Practice: Boolean Values / Logical Operators

### Question 1 — Create two Boolean variables and combine them with `and`.

In [30]:
has_ticket = True
has_id = True

print(has_ticket and has_id)

True


**Line-by-line explanation**
1. `has_ticket` and `has_id` are both `True`.
2. `and` requires **both** to be `True` to give `True` — since both are `True`, the result is `True`.

### Question 2 — Create two Boolean variables and combine them with `or`.

In [31]:
has_cash = False
has_card = True

print(has_cash or has_card)

True


**Line-by-line explanation**
1. `has_cash` is `False`, `has_card` is `True`.
2. `or` only needs **one** side to be `True` — since `has_card` is `True`, the result is `True`.

### Question 3 — Create a Boolean variable and reverse it using `not`.

In [32]:
is_raining = False
print(not is_raining)

True


**Line-by-line explanation**
1. `is_raining` is `False`.
2. `not` reverses a Boolean, so `not False` becomes `True`.

### Question 4 — Create an age condition and an ID condition and combine them with `and`.

In [33]:
age = 20
has_id = True

can_enter = (age >= 18) and has_id
print(can_enter)

True


**Line-by-line explanation**
1. `age >= 18` evaluates to `True` (since `age` is `20`).
2. `has_id` is `True`.
3. `and` combines both conditions — since both are `True`, `can_enter` becomes `True`. This pattern (combining a comparison with a Boolean using `and`) is exactly how real-world access-control logic is written.

### Question 5 — Create two conditions and build one expression using both `and` and `or`.

In [34]:
age = 16
has_guardian = True
has_special_pass = False

can_watch = (age >= 18) or (has_guardian and age >= 13)
print(can_watch)

True


**Line-by-line explanation**
1. `age >= 18` → `False` (age is `16`).
2. `has_guardian and age >= 13` → `has_guardian` is `True` **and** `age >= 13` is `True`, so this combined expression is `True`.
3. The full expression is `False or True`, and since `or` only needs one side `True`, `can_watch` becomes `True`.
4. This shows how `and`/`or` can be combined in one line to express a real rule: "allowed if 18+, OR if accompanied by a guardian and at least 13."

---

## 🎉 Class 3 — All 30 Questions Solved

Every practice question from Class 3 — Strings, User Input, Type Conversion, Arithmetic Operators, Comparison Operators, and Boolean/Logical Operators — has now been solved with code and line-by-line explanations.

---

## 👨‍💻 About This Course

This **free Python course** was designed and developed by **Muhammad Saad** as a contribution toward making programming education more accessible. I created it to share what I’ve learned with students and beginners who want to learn Python but may not have access to paid courses.

The goal is simple: **to make learning Python easier, accessible, and beginner-friendly.** If this course helps someone learn a new skill or take their first step into programming, then this initiative has achieved its purpose.

If you found this class helpful, feel free to share it with someone who might benefit from it.